src/custom/pages/Swap/ArrowWrapperLoader.tsx

```diff
@@ -0,0 +1,136 @@
+import React from 'react'
+import styled, { css } from 'styled-components'
+import loadingCowGif from 'assets/cow-swap/cow-load.gif'
+import useLoadingWithTimeout from 'hooks/useLoadingWithTimeout'
+import { useIsQuoteRefreshing } from 'state/price/hooks'
+
+export interface ArrowWrapperProps {
+  children: React.ReactNode
+}
+
+export function ArrowWrapperLoader({ children }: ArrowWrapperProps) {
+  const COW_LOADING_TIME = 4000
+  const isRefreshingQuote = useIsQuoteRefreshing()
+  const showLoader = useLoadingWithTimeout(isRefreshingQuote, COW_LOADING_TIME)
+
+  return (
+    <Wrapper showLoader={showLoader}>
+      {children}
+      <div>{showLoader && <img src={loadingCowGif} alt="Loading prices..." />}</div>
+    </Wrapper>
+  )
+}
+
+export const Wrapper = styled.div<{ showLoader: boolean }>`
+  position: absolute;
+  background: ${({ theme }) => theme.swap.arrowDown.background};
+  border-radius: ${({ theme }) => theme.swap.arrowDown.borderRadius};
+  width: ${({ theme }) => theme.swap.arrowDown.width};
+  height: ${({ theme }) => theme.swap.arrowDown.height};
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  border: ${({ theme }) => `${theme.swap.arrowDown.borderSize} solid ${theme.swap.arrowDown.borderColor}`};
+  transition: transform 0.1s ease-in-out;
+  padding: 0;
+  cursor: pointer;
+  transform-style: preserve-3d;
+  transform-origin: center right;
+  transition: transform 0.25s;
+
+  &:hover {
+    > svg {
+      stroke: ${({ theme }) => theme.swap.arrowDown.colorHover};
+    }
+  }
+
+  > svg {
+    stroke: ${({ theme }) => theme.swap.arrowDown.color};
+    backface-visibility: hidden;
+    width: 100%;
+    height: 100%;
+    object-fit: contain;
+    padding: 4px;
+    margin: 0;
+    position: absolute;
+
+    ${({ showLoader }) =>
+      showLoader
+        ? css`
```

Comment:
why do you need to use the css function? why not just use template string directly?

Reply:
you may prefer also to just use the `showLoader && cssStylesTemplateString` syntax over the ternary. tho TS might complain about `boolean` not being accepted u can just type it easily

Reply:
is there a reason for this over `visibility: hidden`  or `display: none;`? it's not animating from what i can tell

Reply:
`backface-visibility: hidden;` is used to hide 'the back'. Which you need when you use this on the parent `transform-style: preserve-3d;` 

Essentially creates 'card' which you can flip around and say that the 'back' (backface) should be hidden when you flip it with `transform: rotateY(180deg);`

Reply:
> why do you need to use the css function? why not just use template string directly?
Re-used your code from the GIF loader. Will review now to not need it.

Reply:
Actually without the `css` (ThemedCssFunction) you loose the syntax highlighting...so would propose to keep as is.

Reply:
👍 all good then